packages/worker/src/core.ts

```diff
@@ -58,12 +69,40 @@ export class Core {
           : `http://${compiledServerOptions.address}`,
       },
       telemetryOptions,
+      logger: options.logger ?? new DefaultLogger("INFO")
     };
     const native = await promisify(newCore)(compiledOptions);
+    this.initLogPolling(options, native);
+
     return new this(native, compiledOptions);
   }
 
+  private static initLogPolling(options: CoreOptions, native: native.Core) {
+    if (options.logger != null) {
+      interval(3)
+        .pipe(takeUntil(this.shouldPollForLogs.pipe(filter((v) => v === false))))
+        .subscribe((_) => {
```

Comment:
This doesn't allow catching any errors, let me think about the right way to do this some more, don't have much time left today.

Reply:
So after thinking about this a bit more, the main issue in this implementation apart from the lack of error handling is that nothing is waiting on `corePollLogs`, this means that we might issue more than one concurrent poll.
It'd have been better to use [`concatMap`](https://rxjs-dev.firebaseapp.com/api/operators/concatMap) on the promisified version of `corePollLogs` and make sure that we don't overpoll if lang is falling behind:

```ts
const poll = promisify(corePollLogs);
this.logPollPromise = lastValueFrom(
  of(this.shouldPollForLogs).pipe(
    map((subject) => subject.getValue()),
    takeWhile((shouldPoll) => shouldPoll),
    concatMap(() => poll(native)),
    map((logs) => { /* logging happens here */ }),
    repeat(),
  )
);
this.logPollPromise.reject(() => undefined); // Prevent unhandled rejection
```

Then each worker can subscribe to `logPollPromise` and propagate errors to `the Worker.run` promise.

This is all assuming that errors in logging are all unexpected errors (the process should be aborted).

Reply:
I added this but didn't propagate errors to worker.run because I don't know that there's anything interesting to do with them. I think simply ignoring them is fine, or alternatively just logging an error that we couldn't... log logs.